src/iterators/mod.rs

```diff
@@ -1312,6 +1352,26 @@ impl<'a, A, D: Dimension> AxisChunksIterMut<'a, A, D> {
             life: PhantomData,
         }
     }
+    pub fn split_at(self, index: usize) -> (Self, Self) {
```

Comment:
I see that this 19-lines block is exactly identical to the other `split_at` above. This may be "normal" (I'm not a Rust expert at all), but it looks wrong! Is there any tool to avoid this? Templating `split_chunk_iter` to return the right types?

Reply:
I believe your idea would work, but this is my first contribution to ndarray, so I chose to mirror the existing code style for AxisIter.

Reply:
@nilgoyette I believe it can be done using trait inheritance. `AxisIterMut` can be a sub trait of `AxisIter`

Reply:
I think we can get there with a more lightweight solution: we can have a generic private function and just call it in each of the method. What do you think @nitsky?